Yoast/wordpress-seo
# Add warning component to the sidebar and metabox for catching recalibration errors

## Summary

This PR can be summarized in the following changelog entry:

* Adds a warning message at the top of the metabox and sidebar for when the web worker initialization or analysis fail when using the recalibrated analysis beta.

## Relevant technical choices:

* An analogous error for non-recalibrated analysis will be implemented in https://github.com/Yoast/wordpress-seo/issues/11736

## Test instructions

This PR can be tested by following these steps:

* Link this branch with `11700-handle-worker-error` on YoastSEO.js and with `11700-add-yoast-warning-component` on yoast-components
* If you have an old FlyWheel that throws the original error, don't do update it! Make sure you have the plugin in the FlyWheel location. Tip for a developer: https://github.com/getflywheel/local-addon-volumes -- that way you can symlink your existing plugins directory to the FlyWheel one.
* If you **don't** have an old FlyWheel that throws the original error, fake an error somewhere in the webworker initiation or execution (e.g., [here](https://github.com/Yoast/wordpress-seo/blob/trunk/js/src/analysis/refreshAnalysis.js#L42)).
* Build the plugin with the linked YoastSEO.js/components.
* Go to your SEO/Features and make sure your Recalibration is enabled
* Navigate to a post and observe an error message at the top of the metabox (and if you have Gutenberg enabled, same error message at the top of the sidebar)

## Screenshots

<img width="889" alt="yoastwarning" src="https://user-images.githubusercontent.com/35524806/49450661-c59c7b80-f7dd-11e8-88d2-2a8158ca39d5.png">



## Quality assurance


Fixes #11700
